Task 2

enginire4ek opened the pull request:
- Added a feed of pins on the Home page
- Home page wrapped in SwipeRefreshLayout
- Removed @SuppressLint("RestrictedApi") note when navigating BackStack
- Changed adapter class
- In some XML files, the language of the help information has been changed to Russian

enginire4ek committed:
Task 2
- Added a feed of pins on the Home page
- Home page wrapped in SwipeRefreshLayout
- Removed @SuppressLint("RestrictedApi") note when navigating BackStack
- Changed adapter class
- In some XML files, the language of the help information has been changed to Russian

enginire4ek committed:
Task 2. Updates
- Selected the MVVM design pattern of the application
- Added Repository.kt
- Added HomeViewModel.kt
- Updated states in NetworkState sealed class

102Rudy commented on app/src/main/java/com/example/pinterbest/HomeFragment.kt:
```diff
@@ -29,41 +38,79 @@ class HomeFragment : Fragment() {
+        }
+        if (binding.progressBar.visibility != View.INVISIBLE) {
+            binding.progressBar.visibility = View.INVISIBLE
+        }
```
it's better to use `GONE` instead of `INVISIBLE`

enginire4ek replied:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/2/commits/d8b9e4e644b216cf3291efa52158f7106e897b56

102Rudy commented on app/src/main/java/com/example/pinterbest/adapters/PinFeedHomeAdapter.kt:
```diff
@@ -1,13 +1,20 @@
-class PinFeedHomeAdapter(private val pinObjects: List<PinObject>) :
+class PinFeedHomeAdapter :
     RecyclerView.Adapter<PinFeedHomeAdapter.ViewHolder>() {
+    private var pinObjects: PinsFeed = PinsFeed(listOf())
```
PinsFeed must not be used in UI logic because it is Business model. Consider creating Model especially for using on UI

enginire4ek replied:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/2/commits/d8b9e4e644b216cf3291efa52158f7106e897b56

102Rudy commented on app/src/main/java/com/example/pinterbest/viewmodels/HomeViewModel.kt:
```diff
@@ -0,0 +1,11 @@
+import com.example.pinterbest.data.repository.Repository
+
+class HomeViewModel(application: Application, repository: Repository) :
+    AndroidViewModel(application) {
```
Why do you use `AndroidViewModel` instead of the base one? (I don't remember the name)
Looks like you can use default ViewModel

enginire4ek replied:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/2/commits/d8b9e4e644b216cf3291efa52158f7106e897b56

enginire4ek committed:
Task 2. Updates
- HomeViewModel inherits from ViewModel
- In PinFeedHomeAdapter changed type of pinObjects property
- Create ViewData objects
- Added mapper to ViewData in PinsFeed.kt

102Rudy commented on app/src/main/java/com/example/pinterbest/api/ApiClient.kt:
```diff
@@ -0,0 +1,44 @@
+
+    fun getClient(): Retrofit {
+        val okhttpBuilder = okHttpClient()
+        when (retrofit) {
+            null ->
+                retrofit = Retrofit.Builder()
+                    .baseUrl(BASE_URL)
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .client(okhttpBuilder.build())
+                    .build()
+        }
```
this implementation is not thread-safe

102Rudy commented on app/src/main/java/com/example/pinterbest/adapters/PinObjectViewData.kt:
```diff
@@ -0,0 +1,9 @@
+    var imageLink = pinObject.imageLink
+    var title = pinObject.title
+    var imageAvgColor = pinObject.imageAvgColor
+}
```
Discussed, it's mixing of domains

enginire4ek committed:
Task 2. Updates
- Fixed mapping-function. PinObjectViewData is independent of the PinObject

enginire4ek committed:
Task 2. Updates
- Linter fixes

enginire4ek committed:
Task 2. Updates
- Set up new LayoutManager for RV: StaggeredGridLayoutManager
- Added two new properties imageWidth and imageHeight to PinObjectViewData
- Added dynamic setting of pin height
